tests/sql/test_table_funcs.py

```diff
@@ -1,7 +1,83 @@
+import pytest
+
 import daft
 
 
+@pytest.fixture
+def sample_csv_path():
+    return "tests/assets/mvp.csv"
+
+
+@pytest.fixture
+def sample_schema():
+    return {"a": daft.DataType.float32(), "b": daft.DataType.string()}
+
+
 def test_sql_read_parquet():
     df = daft.sql("SELECT * FROM read_parquet('tests/assets/parquet-data/mvp.parquet')").collect()
     expected = daft.read_parquet("tests/assets/parquet-data/mvp.parquet").collect()
     assert df.to_pydict() == expected.to_pydict()
+
+
+def test_sql_read_csv():
+    df = daft.sql("SELECT * FROM read_csv('tests/assets/mvp.csv')").collect()
+    expected = daft.read_csv("tests/assets/mvp.csv").collect()
+    assert df.to_pydict() == expected.to_pydict()
+
+
+@pytest.mark.parametrize("has_headers", [True, False])
+def test_read_csv_headers(sample_csv_path, has_headers):
+    df1 = daft.read_csv(sample_csv_path, has_headers=has_headers)
+    df2 = daft.sql(f"SELECT * FROM read_csv('{sample_csv_path}', has_headers => {str(has_headers).lower()})").collect()
+    assert df1.to_pydict() == df2.to_pydict()
+
+
+@pytest.mark.parametrize("delimiter", [","])
+def test_read_csv_delimiter(sample_csv_path, delimiter):
+    df1 = daft.read_csv(sample_csv_path, delimiter=delimiter)
+    df2 = daft.sql(f"SELECT * FROM read_csv('{sample_csv_path}', delimiter => '{delimiter}')").collect()
+    assert df1.to_pydict() == df2.to_pydict()
+
+
+@pytest.mark.parametrize("double_quote", [True, False])
+def test_read_csv_quote(sample_csv_path, double_quote):
+    df1 = daft.read_csv(sample_csv_path, double_quote=double_quote)
+    df2 = daft.sql(
+        f"SELECT * FROM read_csv('{sample_csv_path}', double_quote => {str(double_quote).lower()})"
+    ).collect()
+    assert df1.to_pydict() == df2.to_pydict()
+
+
+def test_read_csv_schema(sample_csv_path, sample_schema):
+    df1 = daft.read_csv(sample_csv_path, schema=sample_schema)
+    schema_for_sql = {k: str(v) for k, v in sample_schema.items()}
+    df2 = daft.sql(f"SELECT * FROM read_csv('{sample_csv_path}', schema => {schema_for_sql})").collect()
+    assert df1.to_pydict() == df2.to_pydict()
+
+
+@pytest.mark.parametrize(
+    "escape_char, comment, allow_variable_columns, file_path_column, hive_partitioning, use_native_downloader",
+    [("\\", "#", True, "filepath", False, True)],
+)
+def test_read_csv_other_options(
```

Comment:
since the parameters are a list with a single value, we could probably just pass them in directly instead of `parametrize`